ROVU/Core/src/utility/TestSuite.java

```diff
@@ -53,32 +68,181 @@ public void robotPositioningTest() {
+			Environment testEnv =  new Environment(0.5, etest);
+			testEnv.generateEmptyGrid(40, 0.5);
+			Assert.assertTrue(testEnv.getMap().size()==1600);
+		}
+		
+		@Test
+		/**
+		 * Test #5: Environment grid generation with walls and boundaries
+		 */
+		public void wallBoundaryTest(){
+			EnvironmentDescription etest = new EnvironmentDescription();
+			Environment testEnv =  new Environment(0.5, etest);
+			
+			List <Boundary> testBounds = new ArrayList <Boundary>();
+			Boundary test1 = new HorizontalBoundary(-5.0f, -3.0f, 3.0f, etest, Color.BLUE);
+			Boundary test2 = new VerticalBoundary(-8.0f, -5.7f, 0.1f, etest, Color.GREEN);
+			testBounds.add(test1);
+			testBounds.add(test2);
+			
+			List <Wall> testWalls = new ArrayList <Wall>();
+			Wall test3 = new HorizontalWall(-3.5f,-7.0f,-6.1f,etest,Color.GRAY);
+			Wall test4 = new VerticalWall(8.3f,-9.2f,-4.0f,etest,Color.MAGENTA);
+			testWalls.add(test3);
+			testWalls.add(test4);
+				
+			testEnv.setWalls(testBounds, testWalls);
+			testEnv.generateEmptyGrid(40, 0.5);
+			
+			for (double i=-3;i<3;i+=0.5){
+				Assert.assertTrue(testEnv.pointNode(new Point (-5.0,i), 0.5).isWall());
+			}
+			for (double i=-6;i<0.5;i+=0.5){
+				Assert.assertTrue(testEnv.pointNode(new Point (i,-8.0), 0.5).isWall());
+			}
+			for (double i=-7;i<-6;i+=0.5){
+				Assert.assertTrue(testEnv.pointNode(new Point (-3.5,i), 0.5).isWall());
+			}
+			for (double i=-9.5;i<-4;i+=0.5){
+				Assert.assertTrue(testEnv.pointNode(new Point (i,8.0), 0.5).isWall());
+			}
+		}
+		
+		@Test
+		/**
+		 * Test #6: Environment grid generation with physical and logical areas
+		 */
+		public void physicalLogicalAreasTest(){
+			EnvironmentDescription etest = new EnvironmentDescription();
+			Environment testEnv =  new Environment(0.5, etest);
+			
+			testEnv.addLogicalArea(new Rectangle2D.Double(-2, -1, 3, 2), "Wifi");
+			testEnv.addPhysicalArea(new Rectangle2D.Double(-9, -3, 4, 3), "Office");
+			testEnv.generateEmptyGrid(40, 0.5);
+			
+			for (double i=-9;i<-5;i+=0.5){
+				for (double z=-3;z<0;z+=0.5){
+					Assert.assertTrue(testEnv.pointNode(new Point (i,z), 0.5).isPhysical() && testEnv.pointNode(new Point (i,z), 0.5).getPhysical().get(0).equalsIgnoreCase("Office"));
+				}
+			}
+			
+			for (double i=-2;i<1;i+=0.5){
+				for (double z=-1;z<1;z+=0.5){
+					Assert.assertTrue(testEnv.pointNode(new Point (i,z), 0.5).isWifi() && testEnv.pointNode(new Point (i,z), 0.5).getLogical().get(0).equalsIgnoreCase("Wifi"));
+				}
+			}
+		}
+		
+		@Test
+		/**
+		 * Test #7: Status of node's neighbors
+		 */
+		public void nodeNeighborTest(){
+			EnvironmentDescription etest = new EnvironmentDescription();
+			Environment testEnv =  new Environment(0.5, etest);
+			testEnv.generateEmptyGrid(40, 0.5);
+			
+			Node testNode = testEnv.getMap().get(47);
+			double tX=testNode.getPoint().getX();
+			double tZ=testNode.getPoint().getZ();
+			int test=0;
+			for (Node neigh:testNode.getNeighbors()){
```

Comment:
What does this for-loop do? I really don't follow here. Add a comment for clarification.